subsys/event_manager/event_manager_shell.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <event_manager.h>
+
+u32_t event_manager_displayed_events;
+
+static int show_events(const struct shell *shell, size_t argc,
+		char **argv)
+{
+	shell_fprintf(shell, SHELL_NORMAL,
+		      "Registered Events:\n");
+
+	for (const struct event_type *et = __start_event_types;
+	     (et != NULL) && (et != __stop_event_types); et++) {
+
+		size_t ev_id = et - __start_event_types;
+
+		shell_fprintf(shell,
+			      SHELL_NORMAL,
+			      "%c %d:\t%s\n",
+			      (event_manager_displayed_events & BIT(ev_id)) ?
+				'E' : 'D',
+			      ev_id,
+			      et->name);
+	}
+
+	return 0;
+}
+
+static void set_event_displaying(const struct shell *shell, size_t argc,
+				 char **argv, bool enable)
+{
+	u32_t evt_mask = 0;
+
+	/* If no IDs specified, all registered events are affected */
+	if (argc == 1) {
+		for (const struct event_type *et = __start_event_types;
+		(et != NULL) && (et != __stop_event_types); et++) {
+
+			size_t ev_id = et - __start_event_types;
+
+			evt_mask |= BIT(ev_id);
+		}
+
+		shell_fprintf(shell,
+			      SHELL_NORMAL,
+			      "Displaying all events %sabled\n",
+			      enable ? "en":"dis");
+	} else {
+		const size_t index_cnt = argc - 1;
+		int event_indexes[index_cnt];
```

Comment:
I would prefer `event_index[argc - 1]` 
i.e. not using tmp variable

Reply:
done